test/e2e/service.go

```diff
@@ -1055,21 +1049,14 @@ func testNotReachable(ip string, port int) {
 		Failf("got port==0 for non-reachability check", url)
 	}
 
-	for t := time.Now(); time.Since(t) < podStartTimeout; time.Sleep(5 * time.Second) {
-		resp, err = httpGetNoConnectionPool(url)
+	expectNoError(wait.Poll(5*time.Second, podStartTimeout, func() (bool, error) {
+		resp, err := httpGetNoConnectionPool(url)
 		if err != nil {
-			break
+			return true, nil
 		}
-		body, err = ioutil.ReadAll(resp.Body)
-		Expect(err).NotTo(HaveOccurred())
 		resp.Body.Close()
-		By(fmt.Sprintf("Got success waiting for non-reachability of %s: %v", url, resp.Status))
-	}
-	if err == nil {
-		Failf("able to reach service %s when should no longer have been reachable: %q body=%s", url, resp.Status, string(body))
-	}
-	// TODO: Check type of error
-	By(fmt.Sprintf("Found (expected) error during not-reachability test %v", err))
+		return false, nil
```

Comment:
Please add a log line here on failure (there are so many logging lines we're removing, so it would be nice to have at least one so that both (a) log output viewers and (b) code readers can really quickly see what happened).